### PR TITLE
utils: add collaboration_name_normalizer

### PR DESCRIPTION
**Pull request by pazembrz**
part of: #1689

### CODE REVIEW

File: inspire_schemas/utils.py [hunk 1]
```diff
@@ -347,6 +347,8 @@
     "journals",
 ]
 
+compiled_collaboration_cleaner_regexp = None
```
Comment (MJedr) [inline inspire_schemas/utils.py:350]: Why do we need this?

Comment (pazembrz) [reply inspire_schemas/utils.py:350]: It's caching regexp to not "compile" it every time we will call `normalize_collaboration_name`

**Review by MJedr — APPROVED** [review]

File: inspire_schemas/utils.py (hunk 1 above)
Comment (michamos) [reply inspire_schemas/utils.py:350]: I don't think you ever write to this variable, so I don't think this does what you're claiming.

Comment (pazembrz) [reply inspire_schemas/utils.py:350]: Ah... You are right! It's my fault... Moving code between repos is not always easy :) Thanks for noticing! I will prepare update in few minutes.

**Pull request merged**